src/main/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticator.java

```diff
@@ -256,6 +266,31 @@ protected String extractSubject(final Claims claims, final RestRequest request)
     	return roles;
     }
 
+    /**
+     * Returns the roles from a nested claim
+     * @param claims the claims
+     * @return the roles as ArrayList
+     */
+    private Object getNestedRoles(Claims claims) {
+        String[] keys = rolesKey.split("\\.");
+        Object value = null;
+
+        for (String key: keys) {
```

Comment:
Please provide an example here as a comment, what this logic do exactly.
Provide some JSON and exaplanation.

Reply:
Basically wanted to see what is the format of values in this case? Is it always map? What if is something different in case of other IDPs?

Reply:
The old functionality is untouched see (https://github.com/opendistro-for-elasticsearch/security/blob/35209bb9012c4b5d51bd4cea87046000b67c72b7/src/test/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticatorTest.java#L208-L264)

Basically now you can use:
For keycloaks representation `roles_key: realm_access.roles` for 
```json
"realm_access": {
    "roles": [
      "elasticadmin",
      "offline_access",
      "devportaluser",
      "uma_authorization",
      "apiadmin",
      "apiuser"
    ]
  }
```
As it was before `roles_key: roles`
```json
"roles": [
    "elasticadmin",
    "offline_access",
    "devportaluser",
    "uma_authorization",
    "apiadmin",
    "apiuser"
  ]
```
You can find the full keycloak token here (note that it contain a workaround with a custom mapper to make it useable with opendistro): https://gist.github.com/volkflo/8ce36aa6861596c10e1092f1b39603c6